### ISSUE TITLE
High memory usage while training with tflite_model_maker

### ISSUE THREAD

**Issue by LQR471814** [tensorflow/tensorflow#55500]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): True
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 / Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.8.0
- Python version: 3.7.13
- CUDA/cuDNN version: CUDA 11.1
- GPU model and memory: Tesla K80 (Colab GPU) 24 GB GDDR5

**Describe the current behavior**

No errors, however upon increasing `batch_size` past 1 or 2 an OOM error is incurred. Even when keeping `batch_size` to 1, OOM occurs when using more than 1 epoch. When using 1 `batch_size` and 1 `epoch`, after execution of epochs, the memory usage grows until eventually crashing. 

**Describe the expected behavior**

No errors, the training ends without crashing.

**Standalone code to reproduce the issue**
[colab](https://colab.research.google.com/drive/1bMpah_j9tyyK_cg3uL0KuFv8BBO3k4Iu?usp=sharing)

**Comment by mohantym** [comment]
Hi @LQR471814 ! The Dataset in the above notebook contains images  of songs from different composers . won't it be more suitable for image classification ? if you are looking for image classification of different compositions , you can put the respective images in respective composer's folder and  follow instructions from [here](https://www.tensorflow.org/lite/tutorials/model_maker_image_classification). Thanks!

**Comment by LQR471814** [comment]
Thanks for your response!
The purpose of the model is to detect the location of musical measures within each score so I used an object detection architecture.
Something like this, sorry I didn't make this clearer in the issue.
![model_example](https://user-images.githubusercontent.com/42160088/161872388-d504d091-58a8-453d-8348-8c8365c772c2.png)

**Comment by mohantym** [comment]
@LQR471814 ! I was getting an Attribute error saying **'NoneType' object has no attribute 'label_map'**  in[ Colab environment](https://colab.sandbox.google.com/gist/mohantym/c5e0e96825428b69867e537e8abb2859/measure-detection.ipynb#scrollTo=dYHaqerGBlwI).   Can you refer to this [document ](https://cloud.google.com/vision/automl/object-detection/docs/create-datasets)to create your Dataset properly (don't see labelled images there)? Thanks!

**Comment by LQR471814** [comment]
Looks like I misconfigured some paths when editing the script for colab, I have fixed it could you try again?

**Comment by mohantym** [comment]
@chunduriv ! Could you please look at this issue? Attaching gist in [2.8](https://colab.sandbox.google.com/gist/mohantym/14308f15f0d78cb86fd0245b3294ffdc/measure-detection.ipynb#scrollTo=_ecaUeQAXoSz) for reference.

**Comment by LQR471814** [comment]
well, I've solved the problem myself, so here's some things to watch out for when training your datasets in terms of memory usage.
1. `model.evaluate` for some reason, this uses a lot of memory. so if you're sure you can skip this step then try doing so.
1. the number of images used in the dataset (usually using more than 400 images will end up causing my script to crash)
1. `batch_size`, it's better if you use a larger batch size coupled with a larger epoch count in terms of general model accuracy. however you can decrease it to free up memory usage.
1. model type, the tflite model maker `object_detector` currently only supports `efficientdet_lite` models, however within this category, note that increasing the level of the `efficientdet_lite` model (ex. `efficientdet_lite0` -> `efficientdet_lite4`) will increase memory usage as well

another thing to watch out for, if you're using a `.csv` file, make sure you provide a label for the data and that the coordinates are normalized from `0-1`. failure to do so will not result in an error, but rather absurd results during inference that will leave you scratching your head for hours on end.
